### PR TITLE
Implement select instruction.

### PR DESCRIPTION
**Pull request by ptersilie**
Depends on https://github.com/ykjit/ykllvm/pull/173.

This completely fixes two more lua tests (files.lua, goto.lua) and moves some others along to the next error.

### CODE REVIEW

**Comment by ptersilie** [comment]
Oh, needs a rebase too. Also have to update ykllvm once that's in.

File: tests/c/select.newcg.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+//     jitstate: stop-tracing
+//     --- Begin aot ---
+//     ...
+//     func main(%arg0: i32, %arg1: ptr) -> i32 {
+//     ...
+//     --- End aot ---
+//     --- Begin jit-pre-opt ---
+//     ...
+//     --- End jit-pre-opt ---
+//     foo
```
Comment (ltratt) [inline tests/c/select.newcg.c:18]: Would I be an idiot if I said this test doesn't seem to have `print("foo")` anywhere in it?

Comment (ptersilie) [reply tests/c/select.newcg.c]: It is I who is the idiot. I pushed a completely not working test (I ran the wrong test to check if things were working :facepalm: ).

Comment (ptersilie) [reply tests/c/select.newcg.c]: See [cb11f06](https://github.com/ykjit/yk/pull/1236/commits/cb11f06b971007fd62e947d27bea055f3eee0d2f)

Comment (ltratt) [reply tests/c/select.newcg.c]: That looks a bit better :)

**Comment by ltratt** [comment]
Please rebase & force push when the ykllvm PR has gone in.

**Comment by ptersilie** [comment]
Squashed and rebased.

**Comment by vext01** [comment]
LGTM

**Comment by ptersilie** [comment]
Ok to squash?

**Comment by vext01** [comment]
please squash.

**Comment by ptersilie** [comment]
Squashed.

**Pull request merged**